src/components/custom-field-input-number/custom-field-input-number.jsx

```diff
@@ -35,28 +24,24 @@ const CustomFieldInputNumber = forwardRef(function CustomFieldInputNumber(props,
       return inputRef.current.validity;
     },
     get value() {
-      return inputRef.current.value;
+      return String(inputRef.current.value);
     },
   }));
 
-  const value = props.value === undefined ? '' : props.value.toString();
-
   return (
-    <AbstractCustomField
+    <Number
       className={props.className}
-      defaultValue={value}
-      disabled={props.disabled}
-      errorText={validationMessage}
+      value={props.value}
+      validationMessage={props.errorText}
       id={props.id}
-      inputRef={inputRef}
       label={props.label}
       max={apiLimits.max}
       min={apiLimits.min}
       name={props.name}
       onBlur={props.onBlur}
-      onChange={onChange}
       placeholder={props.placeholder}
-      readOnly={props.readOnly}
+      readOnly={props.readOnly || props.disabled}
```

Comment:
👌 This is a good way to keep backwards compatibility _but_ allowing ourselves to keep iterating on our components.